cpp/include/Ice/Exception.h

```diff
@@ -6,14 +6,170 @@
 #define ICE_EXCEPTION_H
 
 #include "Config.h"
-#include "IceUtil/Exception.h"
 #include "ValueF.h"
 
+#include <exception>
 #include <memory>
+#include <ostream>
+#include <string>
+#include <vector>
 
 namespace Ice
 {
-    using Exception = IceUtil::Exception;
+    /**
+     * Abstract base class for all Ice exceptions.
+     * \headerfile Ice/Ice.h
+     */
+    class ICE_API Exception : public std::exception
+    {
+    public:
+        /**
+         * Constructs an exception.
+         * @param file The file where this exception is constructed.
+         * @param line The line where this exception is constructed.
+         * @param message The error message returned by what(), or nullptr to use the default message.
```

Comment:
The message will replace the "reason" or "unknown" string we have in many local exceptions.